Windows testing-related changes

brettz9 opened the pull request:
- npm: Add eslint-plugins for non-global inclusion during testing
- npm: Bump ESLint major version (notices during `npm install`)
- npm: Avoid the chmod; trips up Windows and handled by `bin`
- Testing: Prefix exec/spawn paths with Windows-friendly `node`
- Docs: Fix README typo
- ~Fix: Have .gitignore to allow `lib` and add; otherwise, install
   of package.json (which occurs by default without `prepublish` or
   a babel install unless running a local install (in npm < 5) of
   browserify-test) looking for the directory complains that
   script is not found; an `install` script would need to add
   `babel` as a regular dependency~

brettz9 committed:
- npm: Add ESLint plugins for non-global inclusion during testing
- npm: Bump ESLint major version (notices during `npm install`)
- npm: Avoid `chmod`; trips up Windows and handled by `bin`
- Testing: Prefix `exec`/`spawn` paths with Windows-friendly `node`
- Docs: Fix README typo

alekseykulikov commented:
Hey @brettz9 ! Thank you for PR! I would love to merge it, but why have you removed /lib from .gitignore?

alekseykulikov commented:
Hey @brettz9 thank you for PR! But I have a clarification about /lib folder. It generates during module publishing and uploading to npm instead of src. So no babel required. I guess all node modules that use babel do so.
Could you add /lib back to .gitignore and I will merge it.

brettz9 commented:
Gotcha--it is just a pain during development, as I like to reference my own forks in npm pending approval though I understand not doing so is a bit cleaner when ready for a published version (and faster for Git cloners to download)....

For IndexedDBShim, we're currently including the dist files in the repo... I think it makes things more widely accessible to people, e.g., to copy-paste online and utilize certain CDNs (e.g., `rawgit.com`) which merely mirror Github without doing the work of npm building... But I see it is a trade-off... And in any case, I added back `lib` to `.gitignore` and removed the `lib` files, so should be good to go. :)

alekseykulikov commented:
thanks @brettz9! 
yes, `/dist` as a name makes more sense, and maybe limiting to node@6+ could be better solution than transipiling.

brettz9 commented:
But when will browsers finally get ES Modules?? Ugh... :)  I know, it has to be done right...
